### PR TITLE
feat(`launch`): split `MsgEditChain`

### LINKED ISSUE #623: Split `MsgEditChain` into two messages
We currently have in `MsgEditChain`, fields to be edited of two types:
- Information not involved in the logic of genesis generation (campaign ID, metadata)
- Information used for the genesis generation (sources, initial genesis, chain ID)

`MsgEditChain` should be splitted.
The former type of information can be editable at any time.
We should not be able to modify the later after the chain launch has been triggered

Proposed solution:
- Keep `MsgEditChain` name and current behavior for metadata and campaign ID, it keeps some consistency with `MsgEditCampaign` that has the same idea with campaign name and metadata
- Add a new message: ex `MsgEditSourceInformation` that modify source hash, URL, initial genesis and chain ID. Prevent updating after trigger launch

Any idea is welcome for message names ideas

### PR DESCRIPTION
**Pull request by aljo242**
Closes #623

### What does this PR does?

- Add new tx `MsgEditChainSourceInformation` which edits source info
- modifies `MsgEditChain` to only modify campaignID and metadata

### CODE REVIEW

**Comment by lubtd** [comment]
Mmh we should address the issue with simulation tests

**Comment by aljo242** [comment]
> Mmh we should address the issue with simulation tests

ah, adding the check for `launchTriggered` broke the sims.  Fixed now :)

File: x/launch/keeper/msg_edit_chain_source_information.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		return nil, sdkerrors.Wrapf(types.ErrChainNotFound, "%d", msg.LaunchID)
+	}
+
+	if chain.LaunchTriggered {
+		return nil, sdkerrors.Wrapf(types.ErrTriggeredLaunch, "%d", msg.LaunchID)
+	}
+
+	if chain.MonitoringConnected {
+		return nil, sdkerrors.Wrapf(types.ErrChainMonitoringConnected, "%d", msg.LaunchID)
+	}
```
Comment (lubtd) [inline x/launch/keeper/msg_edit_chain_source_information.go:32]: `MonitoringConnected` should imply `LaunchTriggered`. This can be removed

**Review by lubtd — COMMENTED** [review]
Looks legit, doing some experimentation

There is the naming that remains tricky.
Having `MsgEditChain` and `MsgEditChainSourceInformation` can be confusing.
I would rename `MsgEditChainSourceInformation` into `MsgUpdateSourceInformation` but we also have `chain-id` value that can be updated that is not really a source information, we should have it in the message because modification should be prevented after launch triggering.

Since information that can't be changed after trigger are related to launch. I would suggest a message name like `MsgUpdateLaunchInformation`
`MsgEditChain` edits miscellaneous static data related to the chain be not influence the launch
`MsgUpdateLaunchInformation` updates information that are actively fetched for the launch with genesis

File: x/launch/client/cli/tx_update_launch_information.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	flagGenesisChainID = "genesis-chain-id"
+	flagSourceURL      = "source-url"
+	flagSourceHash     = "source-hash"
+	flagDefaultGenesis = "default-genesis"
+)
+
+func CmdUpdateLaunchInformation() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-launch-information [launch-id]",
+		Short: "Update chain genesis information",
```
Comment (lubtd) [inline x/launch/client/cli/tx_update_launch_information.go:25]:
```suggestion
		Short: "Update initial information for the launch of the chain",
```

**Review by lubtd — DISMISSED** [review]

File: x/launch/keeper/msg_update_launch_information_test.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	"testing"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/launch/types"
+	profiletypes "github.com/tendermint/spn/x/profile/types"
```
Comment (Pantani) [inline x/launch/keeper/msg_update_launch_information_test.go:13]:
```suggestion
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/stretchr/testify/require"

	testkeeper "github.com/tendermint/spn/testutil/keeper"
	"github.com/tendermint/spn/testutil/sample"
	"github.com/tendermint/spn/x/launch/types"
	profiletypes "github.com/tendermint/spn/x/profile/types"
```

File: x/launch/client/cli/tx.go [hunk 4]
```diff
@@ -29,6 +29,7 @@ func GetTxCmd() *cobra.Command {
 	cmd.AddCommand(CmdSettleRequest())
 	cmd.AddCommand(CmdTriggerLaunch())
 	cmd.AddCommand(CmdRevertLaunch())
+	cmd.AddCommand(CmdUpdateLaunchInformation())
```
Comment (Pantani) [inline x/launch/client/cli/tx.go:32]:
```suggestion
	cmd.AddCommand(
		CmdCreateChain(),
		CmdEditChain(),
		CmdRequestAddAccount(),
		CmdRequestAddVestingAccount(),
		CmdRequestRemoveAccount(),
		CmdRequestAddValidator(),
		CmdRequestRemoveValidator(),
		CmdSettleRequest(),
		CmdTriggerLaunch(),
		CmdRevertLaunch(),
		CmdUpdateLaunchInformation(),
	)
```

**Comment by lubtd** [comment]
Hey @Pantani, do you approve the PR outside of the two suggestions you made?
If yes, we could apply the suggestions to then merge the PR since @aljo242 is OOO end of this week

**Review by Pantani — APPROVED** [review]

File: x/launch/client/cli/tx.go (hunk 4 above)
Comment (lubtd) [reply x/launch/client/cli/tx.go]: Fine, I just changed the order of txs

**Review by giunatale — APPROVED** [review]

**Pull request merged**